utilities/geocoder.py

```diff
@@ -1,46 +1,55 @@
 """
 Geocoding Utility:
-
 We'll be using this: https://github.com/slimkrazy/python-google-places#python-google-places
 to geocode addresses.
 """
 import os
 
 import pandas as pd
 
-from googleplaces import GooglePlaces, types, lang
-
-YOUR_API_KEY = os.getenv('GOOGLE_PLACES_API')
-
+from geopy import GoogleV3
 
-def geocode(df, api_key=YOUR_API_KEY):
+def geocode(df):
     """Add Geocoded columns to df
-
     Keyword Args:
     df: Dataframe which must have an "address" column with a clean address
     api_key: Google Places API Key
     """
-    google_places = GooglePlaces(api_key)
-    matches = []
+    geolocator = GoogleV3()
+    latitudes = []
+    longitudes = []
 
     # This counter is just for debugging purposes since I don't want to hit the API threshold
     i = 0
     for place in df.address:
         print(place)
         print(i)
-        query_result = google_places.nearby_search(
-            location=place,
-            radius=100
-        )
-        print(query_result.places)
-        matches.append(query_result.places)
+        query_result = geolocator.geocode(place)
+
+        if query_result == None:
+            # address not found
+            # tries a new address by cutting off at the first comma, getting rid of secondary address unit
+            new_address = df.borr_street[i].split(',')[0] + ', ' + df.borr_city[i] + ', ' \
+                          + df.borr_state[i] + ', ' + str(df.borr_zip[i])
+            query_result = geolocator.geocode(new_address)
+
+        if query_result != None:
+            print(query_result.latitude, query_result.longitude)
+            latitudes.append(query_result.latitude)
+            longitudes.append(query_result.longitude)
+        else:
+            print("Address Not Found")
+            latitudes.append(None)
+            longitudes.append(None)
```

Comment:
I wonder if this can be done more efficiently. Like if you pull out the code for geocoding one address as a single function, and then have another function that just calls that function passing in the address and api key could that be more efficient using `apply`?